frontend/src/components/G6Graph.svelte

```diff
@@ -36,127 +36,245 @@
         }
     };
 
-    export const focusNode = (id: any) => {
-        graph.focusItem(id);
-    };
+  export const focusNode = (id: any) => {
+    graph.focusItem(id);
+  };
 
-    export const push = (ev: any) => {
-        ev.date = String(ev.time);
-        graph.addItem("node", ev, false, false);
-        for (const target of ev.edges) {
-            graph.addItem("edge", { source: ev.id, target });
-        }
+  export const push = (ev: any) => {
+    ev.date = String(ev.time);
+    graph.addItem("node", ev, false, false);
+    for (const edge of ev.edges) {
+      graph.addItem("edge", {
+        source: ev.id,
+        target: edge.target,
+        label: edge.type,
+      }); // the type of link is connected to the label of the edge here.
+      edgesToBack(ev); // put all edges attached to the node behond the nodes (needed when using groupByTypes: false);
```

Comment:
this call should be outside the loop, now we are redrawing the graph for each edge

Reply:
Will fix it!

Reply:
Fixed for the next commit (waiting for answers to questions before pushing to have one commit addressing all feedback)